### DISCUSSION TITLE
Custom Computer vision script while flying

### DISCUSSION THREAD

**Discussion by HarrySoteriou** [microsoft/AirSim#4191] [Support Q&A]
I am trying to perform precision landing using RL and I am facing the following issues:

1) Can I run a high quality flight simulator while simultaneously utilizing the computer vision features of Airsim, since computer vision mode and flight mode are separated? 
(I 've seen a possibility of gazebo implementation but then why use Airsim in the first place)

2) Is it possible to import my own object detector model ? I have read through the documentation and have yet to see an example demonstrating this.

Appreciate any feedback.

**Comment by zimmy87** [comment]
1. Using a combination of setting "PhysicsEngineName" to "ExternalPhysicsEngine" (see https://microsoft.github.io/AirSim/settings/#physicsenginename) and the setVehiclePose() API, you can pass position and rotation information from an external flight simulator to AirSim. The advantage of doing this would be to leverage Unreal for rendering graphics. Depending upon the desired flight simulator, you may not have access to the same rendering capabilities as Unreal. You can also use ComputerVision [SimMode](https://microsoft.github.io/AirSim/settings/#simmode) to access the graphics and image APIs without simulating physics
2. You can obtain camera output from the client with our [Image APIs](https://microsoft.github.io/AirSim/image_apis/). You can then perform processing on this output using OpenCV in python. See [this thread](https://github.com/microsoft/AirSim/issues/3793) or search for OpenCV in our issues and discussions to see work others have done with this library.
